### ISSUE TITLE
User can't use GitHub(!) without excessive friction

### ISSUE THREAD

**Issue by RickJWagner** [openshiftio/openshift.io#3251]
Using OSIO outside the quickstart scenario, the user might be coerced into bothersome GitHub workarounds.

Example:  The user wants to try a WildFly Swarm quickstart.  (i.e. '/wildfly-swarm-examples/tree/master/jaxrs/jaxrs-cdi').  The expected series of actions:
- Fork the WF Swarm github repo to user's GitHub account
- Select a quickstart from the repo, target it from OSIO's 'Import Code base' action

The problem:  OSIO's 'Import Code base' doesn't allow any more granular selection than the top-level GitHub repository.  (Wildfly Swarm's repository is composed of a set of 'sub-repositories', a common pattern on GitHub.  EAP, Fuse, etc. does the same.)  
For projects like this, as a workaround the user would have to 're-package' the desired sub-repo (i.e. '/wildfly-swarm-examples/tree/master/jaxrs/jaxrs-cdi') as it's own top-level repository.

Is there a better way?

Diagram below, showing only the top-level repo 

![image](https://user-images.githubusercontent.com/524520/39134109-7973ab7e-46db-11e8-834c-694da2d3fc49.png)

**Comment by RickJWagner** [comment]
This seems like a sizeable inconvenience to real-world users.  
Adding @catrobson  for ux visibility.

**Comment by joshuawilson** [comment]
This looks like a feature request to me. If anyone thinks the original design should do this, please change it to a bug.

**Comment by dlabrecq** [comment]
I don't believe it's just a matter of changing one input field to allow a custom path. The back-end APIs must also support subdirectories. Will have to confirm with @aslakknutsen how the import is performed with the provided GitHub URL. However, I believe the APIs currently expect this to be formatted as a GitHub clone URL, with no subdirectory?

**Comment by aslakknutsen** [comment]
This is essentially asking for support for 'mono repo' type deployments, something we don't really support atm. The current model implies 1 repo, 1 build, 1 app, 1 deployment.

**Comment by RickJWagner** [comment]
Our own repositories (OSIO, Fabric8, Wildfly, etc.) are all composed in such a way that components cannot be extracted for use in OSIO without a fork-and-restructure.

Looking around GitHub, this seems to be true for all sorts of repositories-- individual developers, examples from vendors like Spring, etc. etc.  

Most any book that references source code on GitHub is going to have the code organized by chapter.  This is a problem for OSIO, and a problem for developers trying to learn in a hurry.  They'll have additional friction from OSIO when they have to fork-and-restructure to get their sample code to work with OSIO.

It seems that once a project grows beyond a certain point, then the code is restructured into a parent-and-sub-repositories structure.  At that point, OSIO is no longer usable against that repository.
